app/forms/idv_profile_form.rb

```diff
@@ -0,0 +1,48 @@
+class IdvProfileForm
+  attr_reader :user, :errors, :params
+
+  def initialize(user)
+    @user = user
+    @errors = []
+  end
+
+  def submit(params)
+    @params = params
+    error_required_params if required_params_blank?
+    error_duplicate_ssn if ssn_taken?
+    self
```

Comment:
I will confess I did that as an afterthought to make it easier to get the one-liner in the controller. I will change.